app/src/main/java/com/jerboa/feat/BlurNsfwTypes.kt

```diff
@@ -0,0 +1,28 @@
+package com.jerboa.feat
+
+import android.os.Parcelable
+import androidx.annotation.StringRes
+import com.jerboa.R
+import com.jerboa.datatypes.types.PostView
+import com.jerboa.toEnum
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+enum class BlurNsfwTypes(@StringRes val resId: Int) : Parcelable {
+    DoNotBlur(R.string.app_settings_do_not_blur),
+    BlurEverywhere(R.string.app_settings_blur_everywhere),
+    BlurEverywhereExceptNsfw(R.string.app_settings_blur_everywhere_except_nsfw),
+    BlurOnlyNsfwCommunity(R.string.app_settings_blur_only_nsfw_community);
+}
+
+fun Int.needNsfwBlur(postView : PostView) =
+    this.needNsfwBlur(postView.community.nsfw, postView.post.nsfw)
+
+fun Int.needNsfwBlur(isCommunityNsfw: Boolean, isPostNsfw: Boolean = isCommunityNsfw): Boolean {
```

Comment:
Can it be the enum nsfwTypes instead of Int that you extend?

Reply:
@MV-GH do you shure? I think it's 
blurNSFW.needNsfwBlur(communityView.community.nsfw)
looks better than 
blurNSFW.toEnum<BlurNsfwTypes>().needNsfwBlur(communityView.community.nsfw)

Reply:
Yeah, it's not about looking cleaner. The original pollutes autocompletion for all Int. Instead for just BlurNfwsType